src/PoolBuffer.cpp

```diff
@@ -104,119 +133,163 @@ PoolBuffer::PoolBuffer(int /*sample_rate*/, int channels, int bit_res, int FPP,
     }
     mLastWasGlitch = false;
     mOutgoingCnt   = 0;
-    for (int i = 0; i < mPoolSize; i++) {
+
+    mMaxPoolSize = mPoolSize;
+    for (int i = 0; i < mMaxPoolSize; i++) {
         int8_t* tmp = new int8_t[mBytes];
         mIncomingDat.push_back(tmp);
     }
```

Comment:
```suggestion
    for (int i = 0; i < mMaxPoolSize; i++) {
        mIncomingDat.emplace_back(new int8_t[mBytes]);
    }
```

Let's stay with this approach for now, but add a delete loop to the destructor. I have a solution for this in mind with a std::array that is wrapped in a template class, so that you can do `array[m][n]`.

Reply:
And add `mIncomingDat.resize(mMaxPoolSize);` before the loop, so we don't reallocate all the time.